i6_models/parts/frontend/vgg.py

```diff
@@ -0,0 +1,354 @@
+    """
+
+    def __init__(self, model_cfg: VGG4LayerPoolFrontendV1Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        conv_padding = (
+            model_cfg.conv_padding if model_cfg.conv_padding is not None else _get_padding(model_cfg.conv_kernel_size)
+        )
+        pool_padding = model_cfg.pool_padding if model_cfg.pool_padding is not None else 0
+
+        pool_stride = model_cfg.pool_stride
+        self.time_red = pool_stride[0] if isinstance(pool_stride, tuple) else pool_stride
+
+        self.include_linear_layer = True if model_cfg.linear_output_dim is not None else False
+
+        self.conv1 = nn.Conv2d(
+            in_channels=1,
+            out_channels=model_cfg.conv1_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pool = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_kernel_size,
+            stride=model_cfg.pool_stride,
+            padding=pool_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.conv1_channels,
+            out_channels=model_cfg.conv2_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            stride=model_cfg.conv2_stride,
+            padding=conv_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.conv2_channels,
+            out_channels=model_cfg.conv3_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.conv3_channels,
+            out_channels=model_cfg.conv4_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.activation = model_cfg.activation
+        if self.include_linear_layer:
+            self.linear = nn.Linear(
+                in_features=model_cfg.linear_input_dim,
+                out_features=model_cfg.linear_output_dim,
+                bias=True,
+            )
+
+    def forward(self, tensor: torch.Tensor, sequence_mask: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        T might be reduced to T'' depending on the stride of the layers
+
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: masking tensor of shape [B,T], contains length information of the sequences
+        :return: torch.Tensor of shape [B,T',F']
+        """
+        tensor = tensor[:, None, :, :]  # [B,C=1,T,F]
+
+        tensor = self.conv1(tensor)
+        tensor = self.activation(tensor)
+        tensor = self.pool(tensor)  # [B,C,T',F]
+
+        tensor = self.conv2(tensor)  # [B,C,T",F]
+        tensor = self.activation(tensor)
+
+        tensor = self.conv3(tensor)
+        tensor = self.activation(tensor)
+
+        tensor = self.conv4(tensor)
+
```

Comment:
```suggestion
        sequence_mask = _mask_pool(
            sequence_mask, self.conv1.kernel_size[0], self.conv1.stride[0], self.conv1.padding[0]
        )
        sequence_mask = _mask_pool(
            sequence_mask, self.pool.kernel_size[0], self.pool.stride[0], self.pool.padding[0]
        )
        sequence_mask = _mask_pool(
            sequence_mask, self.conv2.kernel_size[0], self.conv2.stride[0], self.conv2.padding[0]
        )
        sequence_mask = _mask_pool(
            sequence_mask, self.conv3.kernel_size[0], self.conv3.stride[0], self.conv3.padding[0]
        )
        sequence_mask = _mask_pool(
            sequence_mask, self.conv4.kernel_size[0], self.conv4.stride[0], self.conv4.padding[0]
        )

```

`_mask_pool` is defined below.